eng/scripts/Invoke-GeneratAndBuildV2.ps1

```diff
@@ -8,33 +8,78 @@ param (
+    $autorestConfig | Set-Content "config.md"
+    $autorestConfigYaml = Get-Content -Path .\config.md
+    $range = ($autorestConfigYaml | Select-String -Pattern '```').LineNumber
+    if ( $range.count -gt 1) {
+        $startNum = $range[0];
+        $lines = $range[1] - $range[0] - 1
+        $autorestConfigYaml = ($autorestConfigYaml | Select -Skip $startNum | Select -First $lines) |Out-String
+    }
+    Install-Module -Name powershell-yaml -Force -Verbose -Scope CurrentUser
+    Import-Module powershell-yaml
+    $yml = ConvertFrom-YAML $autorestConfigYaml
+    $requires = $yml["require"]
+    if ($requires.Count -gt 0) {
+        $readmeFiles = $requires
+    }
+}
 
-foreach ( $relateReadmeFile in $readmeFiles ) {
-    $readmeFile = $relateReadmeFile -replace "\\", "/"
-    
-    $sdkPath =  (Join-Path $PSScriptRoot .. ..)
-    $sdkPath = Resolve-Path $sdkPath
-    $sdkPath = $sdkPath -replace "\\", "/"
+$generatedSDKPackages = New-Object 'Collections.Generic.List[System.Object]'
 
-    
+$sdkPath =  (Join-Path $PSScriptRoot .. ..)
+$sdkPath = Resolve-Path $sdkPath
+$sdkPath = $sdkPath -replace "\\", "/"
+for ($i = 0; $i -le $readmeFiles.Count - 1; $i++) {
+    $readmeFile = $readmeFiles[$i] -replace "\\", "/"
     $readme = ""
-    if ($commitid -ne "") {
-    if ($repoHttpsUrl -ne "") {
-        $readme = "$repoHttpsUrl/blob/$commitid/$readmeFile"
+    if ( $swaggerDir) {
+        $readme = (Join-Path $swaggerDir $readmeFile)
+    } elseif ( $commitid) {
+        if ($repoHttpsUrl) {
+            $readme = "$repoHttpsUrl/blob/$commitid/$readmeFile"
+        } else {
+            $readme = "https://github.com/$org/azure-rest-api-specs/blob/$commitid/$readmeFile"
+        }
     } else {
-        $readme = "https://github.com/$org/azure-rest-api-specs/blob/$commitid/$readmeFile"
+        Write-Error "No readme File path provided."
+        exit 1
     }
-    } else {
-        $readme = (Join-Path $swaggerDir $readmeFile)
+
+    if ($autorestConfigYaml) {
+        $readmeFiles[$i] = $readme
+        $autorestConfigYaml = ConvertTo-YAML $yml
     }
+    Invoke-GenerateAndBuildSDK -readmeAbsolutePath $readme -sdkRootPath $sdkPath -autorestConfigYaml "$autorestConfigYaml" -downloadUrlPrefix "$downloadUrlPrefix" -generatedSDKPackages $generatedSDKPackages
+}
+
+#update services without readme.md
+[System.Collections.ArrayList] $serviceWithReadme = @()
+foreach( $readme in $readmeFiles) {
+    $service, $serviceType = Get-ResourceProviderFromReadme $readme
+    $serviceWithReadme.Add($service)
+}
+[System.Collections.ArrayList] $inputFileToGen = @()
+foreach( $file in $inputFilePaths) {
+    $file -match "specification/(?<specName>.*)/data-plane|resource-manager"
+    if (!$serviceWithReadme.Contains($matches["specName"])) {
+        $inputFileToGen.Add($file)
+    }
+}
 
-    Invoke-GenerateAndBuildSDK -readmeAbsolutePath $readme -sdkRootPath $sdkPath -generatedSDKPackages $generatedSDKPackages
+if ($inputFileToGen.Count -gt 0 ) {
```

Comment:
Tip: really just need `if ($inputFileToGen)`. If it's null or empty, the condition evaluates as `false` already.

Reply:
updated